CombinatorialGames/Surreal/Birthday.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Aaron Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Liu
+-/
+import CombinatorialGames.Game.Birthday
+import CombinatorialGames.Surreal.Ordinal
+
+universe u
+noncomputable section
+
+namespace Surreal
+open IGame NatOrdinal Order Set
+
+/--
+The birthday of a surreal number is defined as the least birthday
+among all *numeric* pre-games that define it. -/
+def birthday (x : Surreal.{u}) : NatOrdinal.{u} :=
+  sInf (IGame.birthday '' {c | ∃ _ : Numeric c, mk c = x})
+
+theorem birthday_eq_iGameBirthday (x : Surreal) :
+    ∃ (y : IGame) (_ : Numeric y), mk y = x ∧ y.birthday = birthday x := by
+  have h : (IGame.birthday '' {c | ∃ _ : Numeric c, mk c = x}).Nonempty := by
+    rw [image_nonempty]
+    exact ⟨_, _, x.out_eq⟩
+  obtain ⟨c, ⟨cn, rfl⟩, hc⟩ := csInf_mem h
+  exact ⟨c, cn, rfl, hc⟩
```

Comment:
If you're willing to slightly change the theorem declaration, you can golf this down to:
```suggestion
theorem birthday_eq_iGameBirthday (x : Surreal) :
    ∃ (y : IGame), (∃ (_ : Numeric y), mk y = x) ∧ y.birthday = birthday x := by
  refine csInf_mem (image_nonempty.2 ?_)
  exact ⟨_, _, x.out_eq⟩
```
If you don't want to do that, you can also do
```suggestion
theorem birthday_eq_iGameBirthday (x : Surreal) :
    ∃ (y : IGame) (_ : Numeric y), mk y = x ∧ y.birthday = birthday x := by
  simp_rw [exists_and_right]
  refine csInf_mem (image_nonempty.2 ?_)
  exact ⟨_, _, x.out_eq⟩
```